python/lsst/meas/deblender/baseline.py

```diff
@@ -243,25 +243,142 @@ def deblend(footprint, maskedImage, psf, psffwhm,
             clipStrayFluxFraction=0.001, clipFootprintToNonzero=True,
             removeDegenerateTemplates=False, maxTempDotProd=0.5
             ):
-    """!
-    Deblend a single ``footprint`` in a ``maskedImage``.
-
-    Each ``Peak`` in the ``Footprint`` will produce a deblended child.
-
-    psfChisqCut1, psfChisqCut2: used when deciding whether a given
-    peak looks like a point source.  We build a small local model of
-    background + peak + neighboring peaks.  These are the
-    chi-squared-per-degree-of-freedom cuts (1=without and 2=with terms
-    allowing the peak position to move); larger values will result in
-    more peaks being classified as PSFs.
-
-    psfChisqCut2b: if the with-decenter fit is accepted, we then
-    apply the computed dx,dy to the source position and re-fit without
-    decentering.  The model is accepted if its chi-squared-per-DOF is
-    less than this value.
-
-    maxNumberOfPeaks: if positive, only deblend the brightest
-                      maxNumberOfPeaks peaks in the parent
+    """
+    Deblend a parent ``footprint`` in a ``maskedImage``. This involves several mandatory and optional steps:
```

Comment:
I'd put a blank line after `maskedImage`, so that it's the "short summary". The "This involves..." goes in as the "extended description". (Terminology per https://developer.lsst.io/docs/py_docs.html#common-structure-of-docstrings).

And actually, `footprint` and `maskedImage` should (I think) be `Footprint` and `MaskedImageF`, because you're talking about types, not variables.